astropy/io/registry.py

```diff
@@ -77,10 +77,11 @@ def get_formats(data_class=None):
         rows.append((format_class[1].__name__, format_class[0], has_read,
                      has_write, has_identify, deprecated))
 
-    data = list(zip(*rows)) if rows else None
+    data = None
+    if rows:
```

Comment:
I like silly stylistic comments. If someone else confirms that this is indeed a bottleneck when importing astropy I'll add an inline comment why this approach was chosen and use your suggestion.

Thank you!